PR: Port website to Lektor-Icon v2 from v0 and remove old unused assets

CAM-Gerlach opened the pull request:
Depends on spyder-ide/lektor-icon#8 ; I'll do the final merge so I can orchestrate everything, once this is approved.

This is the culmination of two years of effort, porting the website over to Lektor-Icon theme V2 from V0, and removes a huge amount of unused, obsolete assets that were used by the previous site before we originally switched to Lektor-Icon years ago.  The look should be very similar on desktop with the current content, but under the hood its basically a total rewrite. It fixes tons of responsiveness issues on mobile/smaller screens, is far more flexible and easily customizable simply by editing the contents.lr or the GUI admin panel, and has lots of new options, templates and flowblocks that we can use going forward. 

I tried to be very diligent to make it easy to review:

* Adding Netlify for previews of both the theme and the website
* Structuring my commits carefully to keep each change atomic, conceptually logical, easy to follow and with the "line noise" stuff up front
* Keeping the website port PR to just strictly technical changes only.

In particular, this PR:
* Didn't change any of the text or image content, even where there were grammar issues or out of date info,
* Carefully minimized the delta to the current version in terms of style and layout even where it will likely be an improvement or neutral (aside from the fixes, responsiveness enhancements and unavoidable changes in the theme)
* Avoided actually enabling most of the new theme features where not strictly necessary.

Once this gets merged, Juanita and I will be able to follow up with fixing the other issues on this repo and that we've discussed, copyediting the text, addressing Isabella's comments, and enabling some of the new features of the theme. With minimal effort, this will allow us to do the following, among others:

* Adding links to the docs for each pane discussed in the "components" section
* Eliminate Spyder-Reports from the plugins list
* Make the navbar link length shorter
* Ddd a donate link to the global navbar
etc.

To make sure that can happen as speedily as possible and avoid merge conflicts, we can address any bugs and regressions vs. the current site here, and make any content-level changes on followup PRs (feel free to let me know here if you like, and I can open an issue or otherwise ensure it gets on the todo list).

Fixes #34
Fixes #56
Fixes #117
Fixes #165
Fixes #174
Fixes #176

It also enables easy followup resolution of #134 and #149

juanis2112 reviewed (changes requested):
Thanks for taking care of @CAM-Gerlach. During our review we found the following issues:

1. The space between Spyder logo and Home is too big compared to how it was before
<Details>
Before:

![Screenshot 2020-09-09 at 4 21 17 PM](https://user-images.githubusercontent.com/18587879/92657697-e0592980-f2ba-11ea-8081-0f625d6f12d3.png)
After:

![Screenshot 2020-09-09 at 4 20 29 PM](https://user-images.githubusercontent.com/18587879/92656487-c0c10180-f2b8-11ea-9dc7-d8413e1d4cc9.png)</Details>
2. Font size of nav bar is too big. It should match the docs font.
3. Font color of body is darker than before
<Details>
Before:

![Screenshot 2020-09-09 at 4 25 58 PM](https://user-images.githubusercontent.com/18587879/92656747-31681e00-f2b9-11ea-8ad7-61d63c9c6cfa.png)

After:

![Screenshot 2020-09-09 at 4 26 03 PM](https://user-images.githubusercontent.com/18587879/92656755-36c56880-f2b9-11ea-9951-48e12b27888f.png).  
</Details>
4. All the images (svgs) including open collective, pane icons, sponsor and donation images resize and appear too big when the page width is less 1200
<Details>
Before:

![Screenshot 2020-09-09 at 4 27 36 PM](https://user-images.githubusercontent.com/18587879/92656853-66747080-f2b9-11ea-909b-96c56cefec91.png)

After:

![Screenshot 2020-09-09 at 4 27 15 PM](https://user-images.githubusercontent.com/18587879/92656868-6b392480-f2b9-11ea-8e4e-d8bce668d066.png)
</Details>
5. Download Spyder with Anaconda font is too big
<Details>
Before:

![Screenshot 2020-09-09 at 4 28 12 PM](https://user-images.githubusercontent.com/18587879/92656918-7ee48b00-f2b9-11ea-8063-d45e94cec157.png)

After:

![Screenshot 2020-09-09 at 4 28 09 PM](https://user-images.githubusercontent.com/18587879/92656927-8310a880-f2b9-11ea-82b1-07b8ceed0f9e.png)
</Details>
6. Font in logo is different than it was. The logo should match the one in the docs page
<Details>
Webpage:

![Screenshot 2020-09-09 at 4 29 02 PM](https://user-images.githubusercontent.com/18587879/92656991-a0457700-f2b9-11ea-85f3-15741e1538f2.png)

Docs page:

![Screenshot 2020-09-09 at 4 29 10 PM](https://user-images.githubusercontent.com/18587879/92657011-a5a2c180-f2b9-11ea-9f91-b1e6094b4dfe.png)
</Details>
7. The top screenshot is not scaling correctly when changing the page width. It overflows when reducing the page width to the least value.
<Details>

![Screenshot 2020-09-09 at 4 30 18 PM](https://user-images.githubusercontent.com/18587879/92657085-c79c4400-f2b9-11ea-8da6-5d3288bfa431.png)
</Details>
8. Background color in the footer is darker than the current page

9. There should only be three lines in the footer (1. Licenses, 2. Resources, 3. Contributors)

10. The font in the footer is thicker and bigger than it was before.
<Details>
Before:

![Screenshot 2020-09-09 at 4 30 52 PM](https://user-images.githubusercontent.com/18587879/92657208-07632b80-f2ba-11ea-9a3d-cb5a90e176de.png)

After:

![Screenshot 2020-09-09 at 4 31 16 PM](https://user-images.githubusercontent.com/18587879/92657212-09c58580-f2ba-11ea-9b51-d0e89ee062a5.png)
</Details>

CAM-Gerlach commented:
In addition to the below, I also implemented the change discussed on spyder-ide/lektor-icon#8, adding a new alternate accent color and shifting the button colors down accordingly, as well as fixing a few other minor issues with the buttons and elsewhere and doing a bit more cleanup.

> 2, 4, and 5. sizing differences

Changed.

> 3 and 8 Dark FG color

Changed, followup recommended to conform to accessibility standards.

<details>

Do note that the slightly darker default color was carefully selected to meet minimum contrast requirements for accessibility that the previous color fails, as @isabela-pf has brought up previously.

</details>

> 6. Font in logo is different than it was. The logo should match the one in the docs page 

Font already identical to old site; just missing small caps (now added). Followup recommended to make logo match docs page,

<details>

The font's identical to the old website; the only difference was the lack of small caps, which I added. The docs site uses a different, raster version; if we want to change to that, I can easily do so (along with removing the `| Blog` via a theme option I'll need to add) in a followup PR.

Comparison of logotype for (L to R) docs, old website, new website

![image](https://user-images.githubusercontent.com/17051931/93028872-8fa84000-f5dc-11ea-89b7-9102d98ae48e.png)

</details>

> 7. The top screenshot is not scaling correctly when changing the page width. It overflows when reducing the page width to the least value. 

Behavior is ~identical or less severe than current website, followup recommended to further avoid clipping.

<details>

The behavior in question is essentially identical to the current website at horizontal resolutions > 768px (desktop) and overflows far less or not at all below that (mobile). The basic scaling mode is a fundamental design feature of the original website theme, and for now I closely replicated the current layout, but I can submit a followup PR making the screenshot smaller to minimize clipping.

Comparison at 800px (nearly as narrow as it gets on desktop):

![image](https://user-images.githubusercontent.com/17051931/93026156-97aab480-f5c9-11ea-900b-b0cf92afdce5.png)

Comparison on a simulated iPhone 6/7/8 Plus:

![image](https://user-images.githubusercontent.com/17051931/93026320-d2611c80-f5ca-11ea-9d9a-51f133d0c362.png)

</details>

> 10. The font in the footer is thicker and bigger than it was before.

Already identical; apparent difference was only due to item 8.

<details>

I did notice the line spacing was 4% larger (which I rectified), but inspection confirms the font face, color and weight are precisely identical for both the smaller and larger footer lines and both bold and normal portions, and the font sizes match to 0.01em/~1% (due to px-em conversion), The heavier appearance was solely due to greater contrast with the background (due to 8.); they now look identical.

![image](https://user-images.githubusercontent.com/17051931/93028337-d98f2700-f5d8-11ea-9916-b0ba64cc8bd0.png)

</details>

> 9. There should only be three lines in the footer (1. Licenses, 2. Resources, 3. Contributors)

Not possible for legal and technical reasons. Added Spyder IDE and cut down mine.

<details>

The component icons credit is on a separate line because its injected site-side by a theme setting, and the extra line for the additional primary author credits required by the CC-BY license (to which I added Spyder IDE) is not modifiable site-side (by design, to avoid license violations) and is needed to even out the line lengths and avoid layout problems at narrow widths/mobile.

</details>

> 1. The space between Spyder logo and Home is too big compared to how it was before

Not currently practical for technical reasons, followup possible.

<details>

This is because the navbar items are aligned right rather than left, which was a necessary consequence of the long-ago overhaul to fix the serious navbar layout problems that you have noticed (despite multiple attempts to do so while preserving the old alignment). Its possible that @dalthviz could come up with a better approach that allows this if given enough time, but given the apparent complexity of the problem relative to the cosmetic style change requested, its way out of scope for this PR.

</details>

ccordoba12 commented:
> Not currently practical for technical reasons, might be possible to address in future.

Could you explain why it's really not possible to reduce this space? @juanis2112, @dalthviz and I agreed that we'd really like to see that space reduced.

CAM-Gerlach commented:
@ccordoba12 Just to make sure before I type it out again, you've read the brief description under the `<details>` tag and you want something more thorough, right? Ironically, I initially included a much more complete technical description of the major complexities that item entailed, my many attempts to retain the current alignment when I originally implemented it years ago, and some potential (though hacky, complex and/or uncertain) avenues for @dalthviz to explore to potentially do so in the future. However, I edited it down to only a brief summary since I didn't want to overwhelm you with details haha.

CAM-Gerlach commented:
Also, now that spyder-ide/lektor-icon#8 is merged, we're committed to merge this with no further theme-side changes (unless I open another PR and re-link it here), and I dropped the commit changing the submodule branch to mine so this can be merged whenever its ready.

ccordoba12 commented:
> Just to make sure before I type it out again, you've read the brief description under the <details> tag and you want something more thorough, right?

I'm very sorry, I just missed that part.

Given that there are serious problems with trying to left-align the navbar (that's what I wanted to know), I think we should merge this one, so you can proceed with the follow up work you mentioned.

@juanis2112, @dalthviz, what do you think?

ccordoba12 reviewed (approved):
Thanks a lot @CAM-Gerlach! The new website is awesome!

dalthviz commented on spyder_website.lektorproject:
```diff
@@ -1,69 +1,55 @@
+nav_extralinks = "Docs: https://docs.spyder-ide.org/"
+footer_links = "Github: https://github.com/spyder-ide/spyder, Twitter: https://twitter.com/spyder_ide, Facebook: https://www.facebook.com/SpyderIDE/, Google Groups: https://groups.google.com/group/spyderlib, Instagram: https://instagram.com/spyderide"
+footer_license_name = "MIT and others"
+footer_license_link = "https://github.com/spyder-ide/spyder-website/blob/master/NOTICE.txt"
```
Is this the right link? (`spyder-website` -> `website-spyder`)?

CAM-Gerlach replied:
Great catch, thanks @dalthviz As you already noticed some hours ago, I rebased the change in and also did a bit of reordering and cleanup on the last few commits.

In order to catch this in the future, along with other regressions, I will add some automated pre-commit hooks that can be run on Travis or locally to check for, and in most cases automatically fix, broken links, whitespace/EoL/tab/etc. issues, unoptimized PNGs, and HTML/CSS validation errors to avoid future regressions/bugs and handle a variety of menial tasks automatically. I could also add eslint at some point in the future, though I won't add it here considering my JS isn't the cleanest and I don't want to block everything else. Thanks!

dalthviz reviewed (commented):
Thanks @CAM-Gerlach I left a simple comment. Besides that this LGTM :+1:

dalthviz reviewed (approved):
Thanks @CAM-Gerlach LGTM :+1:

CAM-Gerlach commented:
In response to the issue @dalthviz caught and I didn't, I'm adding some automated [pre-commit](https://github.com/pre-commit/pre-commit) checks for common issues, to warn for (and in many cases, automatically fix) regressions that this PR fixes as well as easy to introduce and hard to catch errors, like broken links, forgotten merge conflicts, and HTML/CSS/config file errors, as well as handle a variety of menial tasks automatically like optimizing PNGs and fixing whitespace/EoL/tab/etc. issues,.

They'll be able to run on both our CIs when you push and locally when you commit. For the latter, you can simply install the `pre-commit` package with pip or conda, type `pre-commit install`, and it'll take care of the rest (it'll install and configure everything it needs, even binary executables and other packages for languages, in an isolated environment with zero intervention, on all common platforms; the initial install could take a few minutes but after that it should only take a few seconds per commit). I should have it ready by later today, so please hold off on merging until then (in case Juanita approves). Thanks.

ccordoba12 commented:
That sounds like a very good idea @CAM-Gerlach, but I think it's out of the scope for this PR. Could we leave it for another one?

juanis2112 reviewed (approved):
Thanks for taking care of this @CAM-Gerlach. I think this is ready too. I agree with Carlos. I don't think the pre commit checks are necessary for this PR. 

Some comments for the follow-up PRs:

-I agree that we should change the font color based on Isabela's suggestion.
-I like better the logo of the docs page than the current website one so if @ccordoba12 and @dalthviz agree maybe we should use that one.
-I still think the credit section is too long at 5 lines so maybe we can see a better way to put it.
-I think we should definitely fix the space mentioned in number 1 in another PR.

CAM-Gerlach commented:
:+1:

ccordoba12 commented:
Since everyone agress now, let's ago ahead and merge this one.

Thanks a lot @CAM-Gerlach for all your hard work to improve our website!

CAM-Gerlach commented:
I apologize for the delay on this, didn't realize I never actually replied.

> -I agree that we should change the font color based on Isabela's suggestion.

Sorry if there was any confusion; Isabella didn't choose the specific color; rather, she has just previously cited the same accessibility standards that motivated it. It might be a good idea to ask her opinion on the specific colors at some point.

> -I like better the logo of the docs page than the current website one so if @ccordoba12 and @dalthviz agree maybe we should use that one.

Sure, @dalthviz originally implemented the docs one and @ccordoba12 mentioned this to me as well to do, so I'll go ahead with that as requested (its a one-line change site-side but it needs a few changes theme-side to add an option to remove the `| Blog`).

> -I still think the credit section is too long at 5 lines so maybe we can see a better way to put it.

@ccordoba12 brought this up earlier on spyder-ide/lektor-icon#8 and while I proposed a way to get it down to four (as mentioned, three is not possible because the icon credit has to be added site-side via a theme option), he and @dalthviz agreed on keeping the current version. However, I'm fine with implementing that proposal, if you all prefer.

We could also do something more drastic, like collapsing the theme-side block into one line and adding a bunch of spans and media queries to make it responsive. However, best-case that would still take up the same number of lines on mobile and smaller screens, where space is most critical, and where due to the responsiveness improvements it takes less than half the vertical space (<150 px vs >300 px) of the old site despite including two more lines of text.

> -I think we should definitely fix the space mentioned in number 1 in another PR.

I also prefer the left-alignment, but as mentioned I exhausted the limit of my abilities trying to find a solution for this that didn't introduce layout problems or other more serious issues and didn't break responsiveness to different window size and menu content. This is likely going to be a more complex and involved change than it may appear on the surface, but if you think its worth it and want @dalthviz to take a look at it, I'm all for it.
